src/utilities/requests.ts

```diff
@@ -149,22 +149,28 @@ export async function reqHasura<T = any>(
     const defaultError = 'An unexpected error occurred';
     const [error] = json.errors;
     const code = error?.extensions?.code;
+    let errorMessage = error?.extensions?.internal?.error?.message ?? error?.message ?? defaultError;
+
+    // Append a cause if it exists.
+    if (error?.extensions?.cause !== null) {
+      errorMessage += ` - ${error.extensions.cause}`;
```

Comment:
Do we foresee the `cause` to be potentially really long like a stack trace? If we stick the cause on to the main message, we could end up with a really long and unreadable error message. The error console already has support to display longer messages as a collapsible part of the error.  An `Error` constructor does take a `cause` argument in its options that can be used instead.  If `cause` is meant to potentially be really long, then I'd say we try to not make it a part of the main error message.

Reply:
I agree - I think they could be long, and we should leave ourselves room to deal with them differently in the UI - even if that means we need a separate UI change to get this to actually show `cause` in our error panel. I'd be in favor of constructing the Error with `{cause: error.extensions.cause}`